src/kbmod/reprojection.py

```diff
@@ -391,7 +474,80 @@ def _get_first_psf_at_time(work_unit, time):
     return images[index].get_psf()
 
 
-def _reproject_images(science_images, variance_images, mask_images, obstimes, common_wcs, original_wcs):
+def _load_images_and_reproject(
+    file_paths, indices, obstime, obstime_index, common_wcs, original_wcs, directory, filename
+):
+    """Load image data from `WorkUnit` shards.
+
+    Parameters
+    ----------
+    file_paths : `list[str]`
+        List of strings comtaining the images to be reprojected and stitched.
+    inidces : `list[int]`
+        List of `WorkUnit` indices corresponding to the original positions
+        of the images within the `ImageStack`.
+    obstime : `float`
+        observation times for set of images.
+    obstime_index : `int`
+        the index of the unique obstime.
+        i.e. the new index of the mosaicked image in
+        the `ImageStack`.
+    common_wcs : `astropy.wcs.WCS`
+        The WCS to reproject all the images into.
+    original_wcs : `list[astropy.wcs.WCS]`
+        The list of WCS objects for these images.
+    directory : `str`
+        The directory to output the new sharded `WorkUnit`.
+    filename : `str`
+        The base filename for the sharded `WorkUnit`.
+    """
+    science_images = []
+    variance_images = []
+    mask_images = []
+    psfs = []
+
+    for file_path, index in zip(file_paths, indices):
+        with fits.open(file_path) as hdul:
+            science_images.append(hdul[f"SCI_{index}"].data.astype(np.single))
+            variance_images.append(hdul[f"VAR_{index}"].data.astype(np.single))
+            mask_images.append(hdul[f"MSK_{index}"].data.astype(np.single))
```

Comment:
Does it make a difference that this is brought in as a `np.single` data type? Would `np.bool` work?